src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/DynamicEvents/DynamicEvents.cs

```diff
@@ -0,0 +1,129 @@
+            {
+                string name = binder.Name;
+                if (index.TryGetValue(name, out var indexValue))
+                {
+                    // TODO, andrewau, handle the case where we have multiple dynamic event of the same name in a single GC.
+                    //
+                    // That should be an array of these dynamic events
+                    // Somewhere the schema should tell us if this is expected or not so not to generate wrong stuff just because the data is corrupted.
+                    // 
+                    result = new DynamicEvent(indexValue[0], DynamicEventSchema.DynamicEventSchemas.Single(s => string.Equals(s.DynamicEventName, binder.Name)).Fields);
+                    return true;
+                }
+                else
+                {
+                    result = null;
+                    return true;
+                }
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+    }
+
+    internal class DynamicEvent : DynamicObject
+    {
+        private RawDynamic rawDynamic;
+        private Dictionary<string, object> fieldValues;
+
+        public DynamicEvent(RawDynamic rawDynamic, List<KeyValuePair<string, Type>> fields)
+        {
+            this.rawDynamic = rawDynamic;
+            this.fieldValues = new Dictionary<string, object>();
+            int offset = 0;
+            foreach (KeyValuePair<string, Type> field in fields)
+            {
+                object value = null;
+                // TODO, AndrewAu, all types that we envision this will be needed
+                if (field.Value == typeof(ushort))
+                {
+                    value = BitConverter.ToUInt16(rawDynamic.Payload, offset);
+                    offset += 2;
+                }
+                else if (field.Value == typeof(ulong))
+                {
+                    value = BitConverter.ToUInt64(rawDynamic.Payload, offset);
+                    offset += 8;
+                }
+                else
+                {
+                    Debug.Fail("Unknown field type");
+                }
+                this.fieldValues.Add(field.Key, value);
+            }
+        }
+
+        public override bool TryGetMember(GetMemberBinder binder, out object result)
+        {
+            string name = binder.Name;
+            if (fieldValues.TryGetValue(name, out var fieldValue))
+            {
+                result = fieldValue;
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        public override string ToString()
+        {
+            return "I am " + this.rawDynamic.Name + " with these fields: \n" + string.Join("\n", this.fieldValues.Select(kvp => kvp.Key + "->" + kvp.Value));
```

Comment:
Is the "I am" necessary here? Just returning the fields should be sufficient.

Reply:
This is really just telling you "I am a prototype 😄".